app/routines/newflow/create_email_for_user.rb

```diff
@@ -0,0 +1,21 @@
+module Newflow
+  class CreateEmailForUser
+    lev_routine
+
+    protected ###############
+
+    def exec(email:, user:)
+      @email = EmailAddress.create(value: email&.downcase, user_id: user.id)
+
+      # Customize the error message about having an invalid email domain
+      if @email.errors && @email.errors.types.fetch(:value, {}).include?(:missing_mx_records)
+        domain = @email.send(:domain)
+        @email.errors.messages[:value][0] = I18n.t(:"login_signup_form.invalid_email_provider", domain: domain)
```

Comment:
@email.errors.messages[:value][0] looks brittle.  will this always have the 0th element? if this has special contextual meaning, consider remapping it as a private method (that documents why the 0th element).